runtime/centrifuge/src/migrations.rs

```diff
@@ -195,404 +208,316 @@ mod asset_registry {
+						decimals: 6,
+						name: b"USD Coin".to_vec(),
+						symbol: b"USDC".to_vec(),
+						existential_deposit: 1000_u128,
+						location: Some(VersionedMultiLocation::V3(MultiLocation::new(
+							1,
+							X3(Parachain(1000), PalletInstance(50), GeneralIndex(1337)),
+						))),
+						additional: CustomMetadata {
+							mintable: false,
+							permissioned: false,
+							pool_currency: true,
+							// TODO(@review): Should probably be LiquidityPoolsTransferability?
+							transferability: CrossChainTransferability::Xcm(XcmMetadata {
+								fee_per_second: None,
+							}),
+						},
+					},
+				),
+				// Adding LP USDC here
+				(
+					LP_ETH_USDC_CURRENCY_ID,
+					lp_eth_usdc_metadata(
+						LiquidityPoolsPalletIndex::get(),
+						ETHEREUM_MAINNET_CHAIN_ID,
+						ETHEREUM_USDC,
+					),
+				),
+			]
 		}
 
-		#[cfg(feature = "try-runtime")]
-		fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
-			let loc_count = orml_asset_registry::LocationToAssetId::<T>::iter_keys().count() as u32;
-			let meta_count = orml_asset_registry::Metadata::<T>::iter_keys().count() as u32;
-
-			log::info!("Found {} LocationToAssetId keys ", loc_count);
-			log::info!("Found {} Metadata keys ", meta_count);
-
-			let is_centrifuge = meta_count == 7;
-
-			let mut expected_loc_count = 6;
-			let mut expected_meta_count = 7;
-			if !is_centrifuge {
-				expected_loc_count = 2;
-				expected_meta_count = 3;
-			}
-
-			frame_support::ensure!(
-				loc_count == expected_loc_count,
-				"Post: LocationToAssetId Unexpected storage state"
-			);
-
-			frame_support::ensure!(
-				meta_count == expected_meta_count,
-				"Post: Metadata Unexpected storage state"
-			);
-
-			log::info!("AssetRegistryMultilocationToXCMV3: post_upgrade: storage was updated!");
-			Ok(())
-		}
-	}
-}
-/// Returns the count of all keys sharing the same storage prefix
-#[allow(dead_code)]
-pub fn count_storage_keys(prefix: &[u8]) -> u32 {
-	let mut count = 0;
-	let mut next_key = prefix.to_vec();
-	loop {
-		match sp_io::storage::next_key(&next_key) {
-			Some(key) if !key.starts_with(prefix) => break count,
-			Some(key) => {
-				next_key = key;
-				count += 1;
-			}
-			None => {
-				break count;
-			}
+		fn get_catalyst_assets() -> Vec<(
+			CurrencyId,
+			orml_asset_registry::AssetMetadata<Balance, CustomMetadata>,
+		)> {
+			// 02f3a00dd12f644daec907013b16eb6d14bf1c4cb4
+			let gk_bytes: &[u8] = &[
+				2u8, 243u8, 160u8, 13u8, 209u8, 47u8, 100u8, 77u8, 174u8, 201u8, 7u8, 1u8, 59u8,
+				22u8, 235u8, 109u8, 20u8, 191u8, 28u8, 76u8, 180u8,
+			];
+			let mut gk = [0u8; 32];
+			gk[..gk_bytes.len()].copy_from_slice(gk_bytes);
+
+			// 35fd988a3d77251b19d5d379a4775321
+			let tranche_id_bytes = &[
+				53u8, 253u8, 152u8, 138u8, 61u8, 119u8, 37u8, 27u8, 25u8, 213u8, 211u8, 121u8,
+				164u8, 119u8, 83u8, 33u8,
+			];
+			let mut tranche_id = [0u8; 16];
+			tranche_id[..tranche_id_bytes.len()].copy_from_slice(tranche_id_bytes);
+
+			vec![
```

Comment:
Can we also register `Native` here? Wdyt? No hard requirement.

Reply:
Sure! I suppose we want to register all Centrifuge assets actually?

Reply:
I think `Native` is sufficient, no?

Reply:
Not sure TBH. Catalyst is not connected to anything and we control Council. So I would say we can also do it later. Are there any other reasons I am missing? Whatever you think is best and easiest.